### PR TITLE
Containment of segments in a chromosome

### PR DESCRIPTION
**Pull request by ilyaminkin**
#7

### CODE REVIEW

File: GFA-spec.md [hunk 1]
```diff
@@ -139,6 +141,7 @@ The length of the overlap is determined by the `CIGAR` string of the link. When
 | :-----|-------- | :------------- |
 | `RC` | `i` |  Read Coverage  |
 | `NM` | `i` |  Number of mismatches/gaps  |
+| `SRC`| `Z` |  Source of the sequence, e.g. name of the FASTA file containing the chromosome |
```
Comment (lh3) [inline GFA-spec.md]: Every tag has exactly two characters.

Comment (ilyaminkin) [inline GFA-spec.md]: Fixed.

**Comment by sjackman** [comment]
I would remove all references to `chromosome`. It's too specific a term for a generic graph of sequences.

File: GFA-spec.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ The GFA format is a tab-delimited text format for describing a set of sequences
 + **Link** an overlap between two segments. Each link is from the end of one segment to
 the beginning of another segment. The link stores the orientation of each segment and
 the amount of basepairs overlapping.
-+ **Containment** an overlap between two segments where one is contained in the other.
++ **Containment** an overlap between a segment and another segment or a chromosome (for graphs built from complete genomes) where one sequence is contained in the other.
```
Comment (sjackman) [inline GFA-spec.md]: Remove this chunk.

File: GFA-spec.md [hunk 3]
```diff
@@ -121,15 +121,17 @@ The length of the overlap is determined by the `CIGAR` string of the link. When
 
 ## Containment line
 
-(need motivation for this)
+(need motivation for this)?
```
Comment (sjackman) [inline GFA-spec.md]: Remove this chunk.

File: GFA-spec.md [hunk 4]
```diff
@@ -121,15 +121,17 @@ The length of the overlap is determined by the `CIGAR` string of the link. When
...
 
 | Col | Field     |   Type  |   Regexp/Range    |          Brief description |
 |-----|:----------|:------|:-------------------|:-----------------|
 |2  |   `From`      | String |  `[!-)+-<>-~][!-~]*`      | name of segment |
 |3  |   `FromOrient`| String |  `+|-`                    | orientation of From segment |
-|4  |   `To`        | String |  `[!-)+-<>-~][!-~]*`      | name of segment |
-|5  |   `ToOrient`  | String |  `+|-`                    | orientation of To segment |
-|6  |   `pos`       | int    |  `[0-9]*`                 | 0-based start of contained segment |
+|4  |   `To`        | String |  `[!-)+-<>-~][!-~]*`      | name of segment or chromosome |
+|5  |   `ToOrient`  | String |  `+|-`                    | orientation of To segment or a chromosome |
```
Comment (sjackman) [inline GFA-spec.md]: Revert these two lines.

File: GFA-spec.md [hunk 5]
```diff
@@ -139,6 +141,7 @@ The length of the overlap is determined by the `CIGAR` string of the link. When
 | :-----|-------- | :------------- |
 | `RC` | `i` |  Read Coverage  |
 | `NM` | `i` |  Number of mismatches/gaps  |
+| `SC`| `Z` |  Source of the sequence, e.g. name of the FASTA file containing the chromosome |
```
Comment (sjackman) [inline GFA-spec.md]: I'd lean towards `SO` (first two characters) or `SR` (first two consonants), but definitely in bike-shed argument territory here.

File: README.md [hunk 6]
```diff
@@ -15,6 +15,7 @@ The format was first suggested in a [blog post](http://lh3.github.io/2014/07/19/
 + [McCortex](https://github.com/mcveanlab/mccortex)
 + [miniasm](https://github.com/lh3/miniasm)
 + [vg](https://github.com/ekg/vg) 
++ [TwoPaCo](https://github.com/medvedevgroup/TwoPaCo)
```
Comment (sjackman) [inline README.md]: :+1:

**Comment by sjackman** [comment]
Please squash this PR down to a single commit.
See https://github.com/blog/2141-squash-your-commits
and https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

File: GFA-spec.md [hunk 7]
```diff
@@ -88,6 +88,8 @@ The Sequence field can be `*` meaning that the sequence is not stored in the GFA
 | `RC`  | `i`  | Read count     |
 | `FC`  | `i`  | Fragment count |
 | `KC`  | `i`  | k-mer count    |
+| `SH`  | `H`  | SHA-256 checksum of the sequence |
+| `UR`  | `Z`  | URI of the sequence |
```
Comment (sjackman) [inline GFA-spec.md]: Please change to `URI or local file-system path of the sequence`

**Comment by sjackman** [comment]
One last comment (see above) and I'm going to merge after that.
@pmelsted Any comments?

**Comment by sjackman** [comment]
For reference, the wording in the `samtools` spec is
> `UR`: URI of the sequence. This value may start with one of the standard protocols, e.g http: or ftp:. If it does not start with one of these protocols, it is assumed to be a file-system path.

**Comment by sjackman** [comment]
Merged. Thanks for your contribution, Ilya!

**Pull request closed without merging**